empress/support_files/js/empress.js

```
@@ -3792,5 +3792,39 @@ define([
         this.redrawBarPlotsToMatchLayout();
     };
 
+    /**
+     * Returns the col variable value for the node.
+     *
+     * @param {Number} node Postorder position of a node in the tree.
+     * @param {String} col The name of the feature metadata column.
+     *
+     * @return {String} The col variable value for the node or undefined if
+     *                  no value exists.
+     */
+    Empress.prototype.getNodeFeatureMetadataValue = function (node, col) {
+        var colIndx = _.indexOf(this._featureMetadataColumns, col);
+        if (_.has(this._tipMetadata, node)) {
+            return this._tipMetadata[node][colIndx];
+        }
+        if (_.has(this._intMetadata, node)) {
+            return this._intMetadata[node][colIndx];
+        }
+        return undefined;
```

Wait, should this ever happen in practice? At least in the context of `makeFeatureMetadataTable()` I don't think it should, since we've already called `Empress.hasFeatureMetadata()`. I vote we throw an error here instead, unless I am missing something.

---

We can throw an error. I choose to return undefined in case the calling function wants to do something special in situations where no metadata exists (which I guess can also be handled by a try catch block).